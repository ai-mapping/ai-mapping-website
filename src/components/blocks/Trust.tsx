import React from 'react';
import Image from 'next/image';
import Section from '@/components/layout/Section';

// Define types for statistics
type Statistic = {
  value: string;
  description: string;
};

// Define types for partners
type Partner = {
  id: string;
  name: string;
  logo?: string;
};

// Main props interface
export type TrustProps = {
  heading: string;
  statistics: Statistic[];
  partnersTitle: string;
  partners: Partner[];
};

// Default props
export const defaultTrustProps: TrustProps = {
  heading: 'Нам доверяют',
  statistics: [
    {
      value: '5+ лет',
      description: 'опыта поставок запчастей из Китая'
    },
    {
      value: '500+',
      description: 'успешных поставок запчастей'
    },
    {
      value: '98%',
      description: 'довольных клиентов'
    }
  ],
  partnersTitle: 'Наши партнеры в Китае',
  partners: [
    { id: '1', name: 'Партнер 1' },
    { id: '2', name: 'Партнер 2' },
    { id: '3', name: 'Партнер 3' },
    { id: '4', name: 'Партнер 4' }
  ]
};

const Trust = ({
  heading = defaultTrustProps.heading,
  statistics = defaultTrustProps.statistics,
  partnersTitle = defaultTrustProps.partnersTitle,
  partners = defaultTrustProps.partners
}: Partial<TrustProps>) => {
  return (
    <Section background="surface" paddingY="lg" containerWidth="container">
        <div className="bg-background rounded-lg p-8 md:p-12 border border-neutral-dark shadow-lg">
          <h2 className="text-2xl md:text-3xl font-semibold text-center mb-10 text-white">
            {heading}
          </h2>

          <div className="grid grid-cols-1 md:grid-cols-3 gap-8 text-center">
            {statistics.map((stat, index) => (
              <div key={`stat-${index}`} className="bg-surface p-6 rounded-lg shadow-md border border-neutral-dark hover:shadow-lg transition-shadow duration-300">
                <div className="text-4xl font-bold text-primary mb-3">
                  {stat.value}
                </div>
                <p className="text-neutral">{stat.description}</p>
              </div>
            ))}
          </div>

          <div className="mt-12">
            <h3 className="text-xl font-semibold text-center mb-6 text-primary">
              {partnersTitle}
            </h3>
            <div className="flex flex-wrap justify-center items-center gap-8">
              {partners.map((partner) => (
                <div 
                  key={`partner-${partner.id}`}
                  className="w-28 h-20 bg-surface rounded-lg border border-neutral-dark shadow-md flex items-center justify-center hover:shadow-md transition-shadow duration-300"
                >
                  {partner.logo ? (
                    <Image 
                      src={partner.logo} 
                      alt={partner.name} 
                      className="max-w-full max-h-full"
                      width={96}
                      height={64} 
                    />
                  ) : (
                    <span className="font-medium text-sm text-neutral">{partner.name}</span>
                  )}
                </div>
              ))}
            </div>
          </div>
        </div>
    </Section>
  );
};

export default Trust;
