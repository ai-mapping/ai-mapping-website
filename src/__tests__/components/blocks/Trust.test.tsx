import React from 'react';
import { render, screen } from '@testing-library/react';
import Trust, { defaultTrustProps } from '@/components/blocks/Trust';

describe('Trust Component', () => {
  it('renders with default props correctly', () => {
    const { container } = render(<Trust />);
    
    // Check if heading is rendered
    expect(screen.getByText(defaultTrustProps.heading)).toBeInTheDocument();
    
    // Check if all statistics are rendered
    defaultTrustProps.statistics.forEach((stat) => {
      expect(screen.getByText(stat.value)).toBeInTheDocument();
      expect(screen.getByText(stat.description)).toBeInTheDocument();
    });
    
    // Check if partners title is rendered
    expect(screen.getByText(defaultTrustProps.partnersTitle)).toBeInTheDocument();
    
    // Check if all partners are rendered
    defaultTrustProps.partners.forEach((partner) => {
      expect(screen.getByText(partner.name)).toBeInTheDocument();
    });
    
    // Snapshot test
    expect(container).toMatchSnapshot();
  });
  
  it('renders with custom props correctly', () => {
    const customProps = {
      heading: 'Custom Trust Heading',
      statistics: [
        {
          value: 'Custom Value',
          description: 'Custom Description'
        }
      ],
      partnersTitle: 'Custom Partners Title',
      partners: [
        { id: 'custom-1', name: 'Custom Partner' }
      ]
    };
    
    const { container } = render(<Trust {...customProps} />);
    
    // Check if custom heading is rendered
    expect(screen.getByText(customProps.heading)).toBeInTheDocument();
    
    // Check if custom statistic is rendered
    expect(screen.getByText(customProps.statistics[0].value)).toBeInTheDocument();
    expect(screen.getByText(customProps.statistics[0].description)).toBeInTheDocument();
    
    // Check if custom partners title is rendered
    expect(screen.getByText(customProps.partnersTitle)).toBeInTheDocument();
    
    // Check if custom partner is rendered
    expect(screen.getByText(customProps.partners[0].name)).toBeInTheDocument();
    
    // Snapshot test
    expect(container).toMatchSnapshot();
  });
  
  it('renders partners with logos correctly', () => {
    const customProps = {
      partners: [
        { id: '1', name: 'Partner With Logo', logo: '/logo.png' }
      ]
    };
    
    render(<Trust {...customProps} />);
    
    // Check if partner logo is rendered
    const logo = screen.getByAltText(customProps.partners[0].name);
    expect(logo).toBeInTheDocument();
    expect(logo).toHaveAttribute('src', customProps.partners[0].logo);
  });
});