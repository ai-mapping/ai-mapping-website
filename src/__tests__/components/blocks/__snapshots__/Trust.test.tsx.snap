// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trust Component renders with custom props correctly 1`] = `
<div>
  <section
    class="bg-surface py-12"
  >
    <div
      class="container mx-auto px-4 md:px-6 lg:px-8"
    >
      <div
        class="bg-background rounded-lg p-8 md:p-12 border border-neutral-dark shadow-lg"
      >
        <h2
          class="text-2xl md:text-3xl font-semibold text-center mb-10 text-white"
        >
          Custom Trust Heading
        </h2>
        <div
          class="grid grid-cols-1 md:grid-cols-3 gap-8 text-center"
        >
          <div
            class="bg-surface p-6 rounded-lg shadow-md border border-neutral-dark hover:shadow-lg transition-shadow duration-300"
          >
            <div
              class="text-4xl font-bold text-primary mb-3"
            >
              Custom Value
            </div>
            <p
              class="text-neutral"
            >
              Custom Description
            </p>
          </div>
        </div>
        <div
          class="mt-12"
        >
          <h3
            class="text-xl font-semibold text-center mb-6 text-primary"
          >
            Custom Partners Title
          </h3>
          <div
            class="flex flex-wrap justify-center items-center gap-8"
          >
            <div
              class="w-28 h-20 bg-surface rounded-lg border border-neutral-dark shadow-md flex items-center justify-center hover:shadow-md transition-shadow duration-300"
            >
              <span
                class="font-medium text-sm text-neutral"
              >
                Custom Partner
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`Trust Component renders with default props correctly 1`] = `
<div>
  <section
    class="bg-surface py-12"
  >
    <div
      class="container mx-auto px-4 md:px-6 lg:px-8"
    >
      <div
        class="bg-background rounded-lg p-8 md:p-12 border border-neutral-dark shadow-lg"
      >
        <h2
          class="text-2xl md:text-3xl font-semibold text-center mb-10 text-white"
        >
          Нам доверяют
        </h2>
        <div
          class="grid grid-cols-1 md:grid-cols-3 gap-8 text-center"
        >
          <div
            class="bg-surface p-6 rounded-lg shadow-md border border-neutral-dark hover:shadow-lg transition-shadow duration-300"
          >
            <div
              class="text-4xl font-bold text-primary mb-3"
            >
              5+ лет
            </div>
            <p
              class="text-neutral"
            >
              опыта поставок запчастей из Китая
            </p>
          </div>
          <div
            class="bg-surface p-6 rounded-lg shadow-md border border-neutral-dark hover:shadow-lg transition-shadow duration-300"
          >
            <div
              class="text-4xl font-bold text-primary mb-3"
            >
              500+
            </div>
            <p
              class="text-neutral"
            >
              успешных поставок запчастей
            </p>
          </div>
          <div
            class="bg-surface p-6 rounded-lg shadow-md border border-neutral-dark hover:shadow-lg transition-shadow duration-300"
          >
            <div
              class="text-4xl font-bold text-primary mb-3"
            >
              98%
            </div>
            <p
              class="text-neutral"
            >
              довольных клиентов
            </p>
          </div>
        </div>
        <div
          class="mt-12"
        >
          <h3
            class="text-xl font-semibold text-center mb-6 text-primary"
          >
            Наши партнеры в Китае
          </h3>
          <div
            class="flex flex-wrap justify-center items-center gap-8"
          >
            <div
              class="w-28 h-20 bg-surface rounded-lg border border-neutral-dark shadow-md flex items-center justify-center hover:shadow-md transition-shadow duration-300"
            >
              <span
                class="font-medium text-sm text-neutral"
              >
                Партнер 1
              </span>
            </div>
            <div
              class="w-28 h-20 bg-surface rounded-lg border border-neutral-dark shadow-md flex items-center justify-center hover:shadow-md transition-shadow duration-300"
            >
              <span
                class="font-medium text-sm text-neutral"
              >
                Партнер 2
              </span>
            </div>
            <div
              class="w-28 h-20 bg-surface rounded-lg border border-neutral-dark shadow-md flex items-center justify-center hover:shadow-md transition-shadow duration-300"
            >
              <span
                class="font-medium text-sm text-neutral"
              >
                Партнер 3
              </span>
            </div>
            <div
              class="w-28 h-20 bg-surface rounded-lg border border-neutral-dark shadow-md flex items-center justify-center hover:shadow-md transition-shadow duration-300"
            >
              <span
                class="font-medium text-sm text-neutral"
              >
                Партнер 4
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
